sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -148,8 +148,13 @@ def try_split(self, fraction_of_remainder, total_buffer_size):
         # We are "finished" with the (non-existent) previous element.
         current_element_progress = 1
       else:
-        # TODO(SDF): Get actual progress of current element.
-        current_element_progress = 0.5
+        current_element_progress_object = (
+            self.receivers[0].current_element_progress())
+        if current_element_progress_object is None:
+          current_element_progress = 0.5
```

Comment:
Shouldn't we return `0.0` or a special value like `-1` to indicate progress is unknown?

Reply:
We don't return this value, but use it in the computations below. We know we're *somewhere* in the element, so half way is a reasonable choice in the absence of more specific data.